### PR TITLE
Lifecycle updates for dbt Cloud

### PR DESCRIPTION
**Pull request by matthewshaver**
## What are you changing in this pull request and why?

Updating product lifecycle page for dbt Cloud

## Checklist

- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [x] For [docs versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning), review how to [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [x] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding or removing pages (delete if not applicable):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/vercel.json`
- [ ] Run link testing locally with `npm run build` to update the links that point to deleted pages

### CODE REVIEW

File: website/docs/docs/dbt-versions/product-lifecycles.md [hunk 1]
```diff
@@ -9,20 +9,17 @@ dbt Labs is directly involved with the maintenance of two products:
...
-
-- **Deprecated:** Features in this state are not actively worked on or enhanced by dbt Labs and will continue to function as-is until their removal date. 
+dbt Cloud features all fall into one of the following categories:
 
-- **Removed:** Removed features no longer have any level of product functionality or platform support.
+- **Beta:** Beta features are still in development and are only available to select customers. To join a beta, there might be a signup form, or dbt Labs may contact specific customers about testing. Some can be activated by enabling [experimental features](/docs/dbt-versions/experimental-features) in your environment. Beta features are incomplete and might not be entirely stable; they should be used at the customer’s risk, as breaking changes could occur. Beta features might not be fully documented, technical support is limited, and service level objectives (SLOs) might not be provided. Download the [Beta Features Terms and Conditions](/assets/beta-tc.pdf) for more details.
```
Comment (matthewshaver) [inline website/docs/docs/dbt-versions/product-lifecycles.md]:
```suggestion
- **Beta:** Beta features are still in development and are only available to select customers. To join a beta, there might be a signup form, or dbt Labs may contact specific customers about testing. Some can be activated by enabling [experimental features](/docs/dbt-versions/experimental-features) in your account. Beta features are incomplete and might not be entirely stable; they should be used at the customer’s risk, as breaking changes could occur. Beta features might not be fully documented, technical support is limited, and service level objectives (SLOs) might not be provided. Download the [Beta Features Terms and Conditions](/assets/beta-tc.pdf) for more details.
```

File: website/docs/docs/dbt-versions/product-lifecycles.md [hunk 2]
```diff
@@ -9,20 +9,17 @@ dbt Labs is directly involved with the maintenance of two products:
...
+- **Preview:**  Preview features are stable and considered functionally ready for production deployments. Some planned additions and modifications to feature behaviors could occur before they become generally available. New functionality that is not backward compatible could also be introduced. Preview features include documentation, technical support, and service level objectives (SLOs). Features in preview are provided at no extra cost, although they might become paid features when they become generally available.
+- **Generally Available (GA):** Generally available features provide stable features introduced to all qualified production environments. Service level agreements (SLAs) apply to GA features, including documentation and technical support. Certain GA feature availability is determined by the dbt version of the environment. To always receive the most recent GA features, ensure your dbt Cloud [environments](/docs/dbt-cloud-environments) are set to ["Keep on latest version"](/docs/dbt-versions/upgrade-dbt-version-in-cloud#keep-on-latest-version).
```
Comment (matthewshaver) [inline website/docs/docs/dbt-versions/product-lifecycles.md]:
```suggestion
- **Generally Available (GA):** Generally available features provide stable features introduced to all qualified dbt Cloud accounts. Service level agreements (SLAs) apply to GA features, including documentation and technical support. Certain GA feature availability is determined by the dbt version of the environment. To always receive the most recent GA features, ensure your dbt Cloud [environments](/docs/dbt-cloud-environments) are set to ["Keep on latest version"](/docs/dbt-versions/upgrade-dbt-version-in-cloud#keep-on-latest-version).
```

File: website/docs/docs/dbt-versions/product-lifecycles.md [hunk 3]
```diff
@@ -9,20 +9,17 @@ dbt Labs is directly involved with the maintenance of two products:
...
-
-- **Deprecated:** Features in this state are not actively worked on or enhanced by dbt Labs and will continue to function as-is until their removal date. 
+dbt Cloud features all fall into one of the following categories:
 
-- **Removed:** Removed features no longer have any level of product functionality or platform support.
+- **Beta:** Beta features are still in development and are only available to select customers. To join a beta, there might be a signup form, or dbt Labs may contact specific customers about testing. Some can be activated by enabling [experimental features](/docs/dbt-versions/experimental-features) in your account. Beta features are incomplete and might not be entirely stable; they should be used at the customer’s risk, as breaking changes could occur. Beta features might not be fully documented, technical support is limited, and service level objectives (SLOs) might not be provided. Download the [Beta Features Terms and Conditions](/assets/beta-tc.pdf) for more details.
```
Comment (nghi-ly) [inline website/docs/docs/dbt-versions/product-lifecycles.md]:
```suggestion
- **Beta:** Beta features are still in development and are only available to select customers. To join a beta, there might be a signup form or dbt Labs may contact specific customers about testing. Some features can be activated by enabling [experimental features](/docs/dbt-versions/experimental-features) in your account. Beta features are incomplete and might not be entirely stable; they should be used at the customer’s risk, as breaking changes could occur. Beta features might not be fully documented, technical support is limited, and service level objectives (SLOs) might not be provided. Download the [Beta Features Terms and Conditions](/assets/beta-tc.pdf) for more details.
```

File: website/docs/docs/dbt-versions/product-lifecycles.md [hunk 4]
```diff
@@ -9,20 +9,17 @@ dbt Labs is directly involved with the maintenance of two products:
...
+- **Preview:**  Preview features are stable and considered functionally ready for production deployments. Some planned additions and modifications to feature behaviors could occur before they become generally available. New functionality that is not backward compatible could also be introduced. Preview features include documentation, technical support, and service level objectives (SLOs). Features in preview are provided at no extra cost, although they might become paid features when they become generally available.
+- **Generally Available (GA):** Generally available features provide stable features introduced to all qualified dbt Cloud accounts. Service level agreements (SLAs) apply to GA features, including documentation and technical support. Certain GA feature availability is determined by the dbt version of the environment. To always receive the most recent GA features, ensure your dbt Cloud [environments](/docs/dbt-cloud-environments) are set to ["Keep on latest version"](/docs/dbt-versions/upgrade-dbt-version-in-cloud#keep-on-latest-version).
```
Comment (nghi-ly) [inline website/docs/docs/dbt-versions/product-lifecycles.md]:
```suggestion
- **Generally available (GA):** Generally available features provide stable features introduced to all qualified dbt Cloud accounts. Service level agreements (SLAs) apply to GA features, including documentation and technical support. Certain GA feature availability is determined by the dbt version of the environment. To always receive the latest GA features, ensure your dbt Cloud [environments](/docs/dbt-cloud-environments) are set to ["Keep on latest version"](/docs/dbt-versions/upgrade-dbt-version-in-cloud#keep-on-latest-version).
```

**Review by nghi-ly — COMMENTED** [review]
tysm for this! looks great! left a cpl suggs and one question

File: website/docs/docs/dbt-versions/product-lifecycles.md [hunk 5]
```diff
@@ -9,20 +9,17 @@ dbt Labs is directly involved with the maintenance of two products:
...
+- **Generally available (GA):** Generally available features provide stable features introduced to all qualified dbt Cloud accounts. Service level agreements (SLAs) apply to GA features, including documentation and technical support. Certain GA feature availability is determined by the dbt version of the environment. To always receive the latest GA features, ensure your dbt Cloud [environments](/docs/dbt-cloud-environments) are set to ["Keep on latest version"](/docs/dbt-versions/upgrade-dbt-version-in-cloud#keep-on-latest-version).
+- **Deprecated:** Features in this state are no longer being developed or enhanced by dbt Labs and will continue to function as-is until their removal date.
```
Comment (nghi-ly) [inline website/docs/docs/dbt-versions/product-lifecycles.md]: curious, should we say something about "no technical support / not supported" ?

Comment (matthewshaver) [inline website/docs/docs/dbt-versions/product-lifecycles.md]:
```suggestion
- **Deprecated:** Features in this state are no longer being developed or enhanced by dbt Labs. They will continue functioning as-is, and their documentation will persist until their removal date. However, they are no longer subject to technical support. 
```

**Review by nghi-ly — APPROVED** [review]
:shipit:

**Pull request merged**